### ISSUE TITLE
[Bug]: shell.openExternal() does not open the application to the foreground on macOS

### ISSUE THREAD

**Issue by popod** [electron/electron#34861]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

19.0.8

### What operating system are you using?

macOS

### Operating System Version

macOS Monterey (12.4)

### What arch are you using?

x64

### Expected Behavior

When running the following code from a renderer process on macOS, the opened is bring to the foreground ([`shell.openExternal()` doc](https://www.electronjs.org/docs/latest/api/shell#shellopenexternalurl-options)).

```js
shell.openExternal('https://www.electronjs.org')
```

### Actual Behavior

When running the following code from a renderer process on macOS, the opened app does not get focus and is not bring to the foreground.

```js
shell.openExternal('https://www.electronjs.org')
```

It seems that this is working fine when the call is done from the main process. Only the renderer process had this bug.

**Comment by codebytere** [comment]
Are you sure you have settings configured correctly? If i add
```
const { shell } = require('electron')

shell.openExternal('https://www.electronjs.org')
```

to the renderer process with these BrowserWindow options:

```
const mainWindow = new BrowserWindow({
    width: 800,
    height: 600,
    webPreferences: {
      preload: path.join(__dirname, 'preload.js'),
      nodeIntegration: true,
      contextIsolation: false
    }
  })
```

it correctly opens in the foreground.

**Comment by popod** [comment]
Hi @codebytere 

Thank you for your reply and sorry for this issue !

Yes, all works fine in fact..

This was my implementation which prevents from focusing on Safari (bring it to foreground). This seems to be because a field kept the focus on the page...

Sorry for that and thank you for your reply !

**Comment by Josue775** [comment]
I am trying to open links that are within my electron app that are in index.html, but what happens is that it opens the link from the electron application itself, not externally.




function createWindow() {
  win = new BrowserWindow({
    width: 1366,
    height: 768,
    autoHideMenuBar: true,
    icon: iconPath,
    webPreferences: {
      nativeWindowOpen: true,
      sandbox: false,
      preload: path.join(__dirname, "preload.js"),
    },
  });
  win.loadFile("index.html");
  win.webContents.on("will-navigate", (event, navigationUrl) => {
    event.preventDefault();
    shell.openExternal(navigationUrl);
  });
}


The image shows how I open the links externally in Windows and pull it, but on Mac that does not work, it does not open the links externally but instead loads it in the same electron application

**Comment by theogravity** [comment]
I just spent a day and half troubleshooting this. This definitely should be documented somewhere. The limit isn't with the browser, but Windows OS `ShellExecute` having a limit of 2083 chars (in my personal testing, it was 2048).


Related Chromium ticket:

https://bugs.chromium.org/p/chromium/issues/detail?id=727909

```
In Chrome, if you attempt to an Application Protocol URL over 2046 characters, you’ll see the regular Security Prompt, but clicking its “Launch Application” button does nothing.
```

tl;dr: Custom protocols in Windows have a limit of 2048 - 2083 characters. If you have such a URL coming from Edge / Chrome, you'll get the prompt to open the app associated with the protocol, but on confirmation, nothing will get launched.

If you trim the URL length to less than 2048, your electron app will launch.

(You may be asking what are you putting in your custom protocol - for us it was a signed JWT, which can easily exceed that limit. We'll have to implement a device auth grant instead for our electron app.)
